src/utils/SpokeUtils.ts

```diff
@@ -200,10 +205,11 @@ export async function getBlockRangeForDepositId(
  * @param blockTag The block number to search for the deposit ID at.
  * @returns The deposit ID.
  */
-export async function getDepositIdAtBlock(contract: Contract, blockTag: number): Promise<number> {
-  const depositIdAtBlock = await contract.numberOfDeposits({ blockTag });
+export async function getDepositIdAtBlock(contract: Contract, blockTag: number): Promise<BigNumber> {
+  const _depositIdAtBlock = await contract.numberOfDeposits({ blockTag });
+  const depositIdAtBlock = toBN(_depositIdAtBlock);
   // Sanity check to ensure that the deposit ID is an integer and is greater than or equal to zero.
-  if (!Number.isInteger(depositIdAtBlock) || depositIdAtBlock < 0) {
+  if (!BigNumber.isBigNumber(depositIdAtBlock) || depositIdAtBlock.lt(bnZero)) {
```

Comment:
@bmzig Instead of checking `BigNumber.isBigNumber()`, I think we'd want to wrap `toBN()` in a try/catch clause. This handles the case where `depositIdAtBlock` is some unexpected type that can't be converted to a BN.

Reply:
I removed it and didn't add it in a try/catch block since it looks like this function throws errors anyways. Do you think that's fine? https://github.com/across-protocol/sdk/pull/830/commits/20d698c8a610ed4b37e6976a1dd971816338ff7e

Reply:
Yeah it has the same effect and the code is more concise than adding a try/catch clause. It might be harder to debug, but I doubt we'll ever see this in practice.